### PR TITLE
fix(composer): don’t add vendor directory

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Fixes Composer Artifacts updating so that vendor files are only checked and added if the vendor folder previous existed.

## Context:

#7935 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] Unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/fs/index.ts [hunk 1]
```diff
@@ -79,3 +79,12 @@ export async function ensureCacheDir(
...
   return cacheDirName;
 }
+
+export function localPathExists(pathName: string): Promise<boolean> {
+  // Works for both files as well as directories
+  return new Promise<boolean>((resolve) => {
+    fs.stat(join(localDir, pathName), (err) => {
+      resolve(!err);
+    });
+  });
```
Comment (viceice) [inline lib/util/fs/index.ts:89]: Why not use await with try catch?

**Review by viceice — DISMISSED** [review]
Just a Questions, otherwise LGTM

File: lib/util/fs/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/util/fs/index.ts:89]: Wasn't sure if this function supported it

Comment (viceice) [reply lib/util/fs/index.ts:89]: we are using it in gradle-wrapper

https://github.com/renovatebot/renovate/blob/5d99ff9862bac803dcea4e6ded5f3fec34abccfc/lib%2Fmanager%2Fgradle-wrapper%2Fartifacts.ts#L63

File: lib/util/fs/index.ts [hunk 2]
```diff
@@ -79,3 +79,12 @@ export async function ensureCacheDir(
   await fs.ensureDir(cacheDirName);
   return cacheDirName;
 }
+
+export function localPathExists(pathName: string): Promise<boolean> {
+  // Works for both files as well as directories
+  return new Promise<boolean>((resolve) => {
+    fs.stat(join(localDir, pathName), (err) => {
+      resolve(!err);
+    });
```
Comment (viceice) [inline lib/util/fs/index.ts:88]:
```suggestion
  return fs.stat(join(localDir, pathName).then(s=> !!s).catch(() => false);
```

**Review by viceice — APPROVED** [review]
Transient gradle error 😕

**Pull request merged**